File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Set<Integer> seenNumbers = new HashSet<>();
        for (int num: nums) {
            // if its in the set return immediately
            if (!seenNumbers.add(num)) {
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        max_streak = 0

        for num in nums_set:
            if (num - 1) not in nums_set:
                length = 1

                while (num + length) in nums_set:
                    length += 1
                max_streak = max(length, max_streak)
        return max_streak



            

        